File: frontend/src/types/brand-kit.ts
export interface FontsConfig {
  heading?: string;
  body?: string;
  disclaimer?: string;
  heading_url?: string;
  body_url?: string;
  disclaimer_url?: string;
  disclaimer_inherited?: boolean;
  size_scale?: Record<string, Record<string, number>>;
}

export type ZoneColorRole = "primary" | "secondary" | "accent" | "white";

export interface ZoneRoles {
  poster_background?: ZoneColorRole;
  cta_fill?: ZoneColorRole;
  disclaimer_strip?: ZoneColorRole;
  badge_callout?: ZoneColorRole;
  headline_text?: ZoneColorRole;
}

export interface ColorNames {
  primary_name?: string;
  secondary_name?: string;
  accent_name?: string;
  primary_usage?: string;
  secondary_usage?: string;
  accent_usage?: string;
}

export interface BrandKit {
  id: string;
  name: string;
  logo_url: string | null;
  secondary_logo_url: string | null;
  primary_color: string;
  secondary_color: string;
  accent_color: string;
  fonts: FontsConfig | null;
  version: number;
  updated_by: string | null;
  updated_at: string;
  is_active: boolean;
  changelog: string | null;
  activated_by_info: { id: string; name: string } | null;
  activated_at: string | null;
  color_names: ColorNames | null;
  zone_roles: ZoneRoles | null;
}

export interface UpdateBrandKitRequest {
  primary_color?: string;
  secondary_color?: string;
  accent_color?: string;
  fonts?: FontsConfig;
  color_names?: ColorNames;
  zone_roles?: ZoneRoles;
  changelog?: string;
}

export interface BrandKitVersionSummary {
  id: string;
  version: number;
  name: string;
  changelog: string | null;
  activated_by_info: { id: string; name: string } | null;
  activated_at: string | null;
  is_active: boolean;
  created_at: string;
  primary_color: string;
  secondary_color: string;
  accent_color: string;
  logo_url: string | null;
  secondary_logo_url: string | null;
  fonts: FontsConfig | null;
  color_names: ColorNames | null;
  zone_roles: ZoneRoles | null;
}

export interface TemplateZone {
  name: string;
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface BrandKitTemplate {
  id: string;
  name: string;
  layout_key: string;
  zones: TemplateZone[];
  is_default: boolean;
}

File: backend/app/schemas/brand_kit.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class FontsConfig(BaseModel):
    heading: str | None = None
    body: str | None = None
    disclaimer: str | None = None
    heading_url: str | None = None
    body_url: str | None = None
    disclaimer_url: str | None = None
    disclaimer_inherited: bool | None = None
    size_scale: dict | None = None


class ColorNames(BaseModel):
    primary_name: str | None = None
    secondary_name: str | None = None
    accent_name: str | None = None
    primary_usage: str | None = None
    secondary_usage: str | None = None
    accent_usage: str | None = None


class ActivatedByInfo(BaseModel):
    id: uuid.UUID
    name: str


class BrandKitResponse(BaseModel):
    id: uuid.UUID
    name: str
    logo_url: str | None = None
    secondary_logo_url: str | None = None
    primary_color: str
    secondary_color: str
    accent_color: str
    fonts: dict[str, Any] | None = None
    version: int
    updated_by: uuid.UUID | None = None
    updated_at: datetime
    is_active: bool = True
    changelog: str | None = None
    activated_by_info: ActivatedByInfo | None = None
    activated_at: datetime | None = None
    color_names: dict | None = None
    zone_roles: dict | None = None

    model_config = {"from_attributes": True}


class BrandKitVersionSummary(BaseModel):
    id: uuid.UUID
    version: int
    name: str
    changelog: str | None = None
    activated_by_info: ActivatedByInfo | None = None
    activated_at: datetime | None = None
    is_active: bool
    created_at: datetime
    # Kit values — shown in the preview panel before restoring
    primary_color: str
    secondary_color: str
    accent_color: str
    logo_url: str | None = None
    fonts: dict[str, Any] | None = None

    model_config = {"from_attributes": True}


class TemplateZone(BaseModel):
    name: str = Field(min_length=1, max_length=64, pattern=r"^[a-z][a-z0-9_]*$")
    x: int = Field(ge=0)
    y: int = Field(ge=0)
    width: int = Field(gt=0)
    height: int = Field(gt=0)


class BrandKitTemplateResponse(BaseModel):
    id: uuid.UUID
    name: str
    layout_key: str
    zones: list[TemplateZone]
    is_default: bool

    model_config = {"from_attributes": True}


class CreateTemplateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    layout_key: str = Field(min_length=1, max_length=100, pattern=r"^[a-z0-9_]+$")
    zones: list[TemplateZone] = Field(min_length=1)


class UpdateTemplateRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    zones: list[TemplateZone] | None = None


class UpdateBrandKitRequest(BaseModel):
    primary_color: str | None = Field(None, pattern=r"^#[0-9A-Fa-f]{6}$")
    secondary_color: str | None = Field(None, pattern=r"^#[0-9A-Fa-f]{6}$")
    accent_color: str | None = Field(None, pattern=r"^#[0-9A-Fa-f]{6}$")
    fonts: dict[str, Any] | None = None
    color_names: dict | None = None
    zone_roles: dict | None = None
    changelog: str | None = None
